File: _code-samples/permissioned-domains/js/modify-domain.js
import xrpl from 'xrpl'
import { stringToHex } from '@xrplf/isomorphic/dist/utils/index.js'
import fs from 'fs'

// Set up client --------------------------------------------------------------
const client = new xrpl.Client('wss://s.altnet.rippletest.net:51233')
await client.connect()

// Load setup data ------------------------------------------------------------
if (!fs.existsSync('setup.json')) {
  console.error('Config data not found. Did you run create_domain.py first?')
  process.exit(1)
}
const configData = JSON.parse(fs.readFileSync('setup.json', 'utf8'))
const wallet = xrpl.Wallet.fromSeed(configData.owner.seed)
if (wallet.address !== configData.owner.address) {
    console.error('Address did not match saved value. Did you use the wrong ',
        'cryptographic algorithm?',
        '\n\tSaved:', configData.owner.address,
        '\n\tGenerated:', wallet.address)
    client.disconnect()
    process.exit(2)
}

const issuerAddress = configData.issuer_address
const domainID = configData.domain_id

console.log('Domain ID:', domainID)
console.log('Domain owner:', wallet.address)
console.log('Credential issuer:', issuerAddress)

// Modify a permissioned domain -----------------------------------------------
// To demonstrate updating the domain, this tutorial uses a different credential
// type (issued by the same issuer, unless you modified setup.json)
const credentialType = stringToHex("new-credential-type")

const pDomainSet = {
  TransactionType: "PermissionedDomainSet",
  Account: wallet.address,
  DomainID: domainID,
  AcceptedCredentials: [
    {
      Credential: {
        Issuer: issuerAddress,
        CredentialType: credentialType
      }
    }
  ]
}
console.log('Submitting transaction', JSON.stringify(pDomainSet, null, 2))
const response = await client.submitAndWait(pDomainSet, { wallet, autofill: true})
console.log(response)
const resultCode = response.result.meta.TransactionResult
if (resultCode !== 'tesSUCCESS') {
  console.error(`PermissionedDomainSet failed with code ${resultCode}`)
  client.disconnect()
  process.exit(3)
}
console.log('Successfully modified permissioned domain.')
client.disconnect()

File: _code-samples/permissioned-domains/py/modify_domain.py
#!/usr/bin/env python
import json
import os
import sys

from xrpl.clients import JsonRpcClient
from xrpl.models.transactions import PermissionedDomainSet
from xrpl.models.transactions.deposit_preauth import Credential
from xrpl.transaction import submit_and_wait
from xrpl.utils import str_to_hex
from xrpl.wallet import Wallet

# Set up client ---------------------------------------------------------------
client = JsonRpcClient("https://s.altnet.rippletest.net:51234")

# Load setup data -------------------------------------------------------------
if not os.path.exists("setup.json"):
    print("ERROR: Config data not found. Did you run create_domain.py first?")
    sys.exit(1)

with open("setup.json") as f:
    config_data = json.load(f)

wallet = Wallet.from_seed(config_data["owner"]["seed"])
if wallet.address != config_data["owner"]["address"]:
    print("ERROR: Address did not match saved value. Did you use the wrong "
          "cryptographic algorithm?"
          "\n\tSaved:", config_data["owner"]["address"],
          "\n\tGenerated:", wallet.address)
    sys.exit(2)

issuer_address = config_data["issuer_address"]
domain_id = config_data["domain_id"]

print("Domain ID:", domain_id)
print("Domain owner:", wallet.address)
print("Credential issuer:", issuer_address)

# Modify a permissioned domain ------------------------------------------------
# To demonstrate updating the domain, this tutorial uses a different credential
# type (issued by the same issuer, unless you modified setup.json)
credential_type = str_to_hex("new-credential-type")

p_domain_set = PermissionedDomainSet(
    account=wallet.address,
    domain_id=domain_id,
    accepted_credentials=[
        Credential(
            issuer=issuer_address,
            credential_type=credential_type
        )
    ]
)
print("Submitting transaction", p_domain_set)
pds_response = submit_and_wait(p_domain_set, client=client, wallet=wallet)
if pds_response.status != "success":
    print("Transaction submission failed:", pds_response)
    sys.exit(1)
print(json.dumps(pds_response.result, indent=2))
result_code = pds_response.result["meta"]["TransactionResult"]
if result_code != "tesSUCCESS":
    print("Transaction failed with result code", result_code)
    sys.exit(3)
print("Successfully modified permissioned domain.")
